android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/WalletUtils.java

```diff
@@ -46,24 +56,38 @@ public static void promptToInstallWallet() {
     if (act == null) {
       return;
     }
+
+    final Intent appStoreIntent =
+        new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + walletPackageName));
+
     AlertDialog.Builder builder;
     builder = new AlertDialog.Builder(act);
     builder.setTitle(R.string.wallet_missing);
     builder.setMessage(act.getString(R.string.install_wallet_from_iab));
     Log.d("String name: ", act.getString(R.string.install_wallet_from_iab));
 
-    builder.setPositiveButton(R.string.install, new DialogInterface.OnClickListener() {
-      @Override public void onClick(DialogInterface dialog, int which) {
-        act.startActivity(
-            new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + walletPackageName)));
-      }
-    });
+      builder.setPositiveButton(R.string.install, new DialogInterface.OnClickListener() {
+        @Override public void onClick(DialogInterface dialog, int which) {
+          try {
```

Comment:
can the code inside of onClick be split in a different function for an easier code read?